roles/prometheus/server/tasks/config.yml

```
@@ -28,29 +28,31 @@
 - name: Template Prometheus file_sd targets for TDP components
   template:
     src: file_sd.yml.j2
-    dest: "{{ prometheus_static_targets_dir }}/{{ item.component }}.yml"
+    dest: "{{ prometheus_static_targets_dir }}/{{ item.value.component }}.yml"
   vars:
-    hosts: "{{ groups[item.group] | map('tosit.tdp.access_fqdn', hostvars) | list }}"
-    port: "{{ item.port }}"
+    hosts: "{{ groups[item.value.group] | map('tosit.tdp.access_fqdn', hostvars) | list }}"
+    port: "{{ item.value.port }}"
     labels:
-      component: "{{ item.component }}"
+      component: "{{ item.value.component }}"
       cluster: "{{ cluster_name }}"
-      service: "{{ item.service }}"
-  loop: "{{ prometheus_tdp_components_targets }}"
+      service: "{{ item.value.service }}"
+  when: item.value.group in groups.keys()
+  loop: "{{ prometheus_tdp_generated_targets | combine(prometheus_tdp_additional_targets) | dict2items)}}"
```

Can you move the `combine` to `tdp_vars` by setting a variable `prometheus_tdp_components_targets` like this:
```
prometheus_tdp_components_targets: {{ prometheus_tdp_generated_targets | combine(prometheus_tdp_additional_targets) }}
```
And use this variable:
```suggestion
  loop: "{{ prometheus_tdp_components_targets | dict2items)}}"
```
It allows the user to disable TDP generated targets.

---

Wait we should talk about that, I'm not sure how it will be merge.